aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,221 @@
+        with folder.open(input_filename, 'w') as infile:
+            for namelist_name in namelists_toprint:
+                infile.write(u'&{0}\n'.format(namelist_name))
+                # namelist content; set to {} if not present, so that we leave an empty namelist
+                namelist = parameters.pop(namelist_name, {})
+                for key, value in sorted(six.iteritems(namelist)):
+                    infile.write(convert_input_to_namelist_entry(key, value))
+                infile.write(u'/\n')
+
+
+        # Check for specified namelists that are not expected
+        if parameters:
+            raise exceptions.InputValidationError(
+                'The following namelists are specified in parameters, but are '
+                'not valid namelists for the current type of calculation: '
+                '{}'.format(','.join(list(parameters.keys()))))
+
+        remote_copy_list = []
+        local_copy_list = []
+
+        # Copy remote output dir
+        parent_calc_folder = self.inputs.get('parent_folder', None)
+        if isinstance(parent_calc_folder, orm.RemoteData):
+            remote_copy_list.append((
+                parent_calc_folder.computer.uuid,
+                os.path.join(parent_calc_folder.get_remote_path(), self._INPUT_SUBFOLDER),
+                self._OUTPUT_SUBFOLDER
+            ))
+            remote_copy_list.append((
+                parent_calc_folder.computer.uuid,
+                os.path.join(parent_calc_folder.get_remote_path(), self._PSEUDO_SUBFOLDER),
+                self._PSEUDO_SUBFOLDER
+            ))
+        elif isinstance(parent_calc_folder, orm.FolderData):
+            for filename in parent_calc_folder.list_object_names():
+                local_copy_list.append((
+                    parent_calc_folder.uuid,
+                    filename,
+                    os.path.join(self._OUTPUT_SUBFOLDER, filename)
+                ))
+                local_copy_list.append((
+                    parent_calc_folder.uuid,
+                    filename,
+                    os.path.join(self._PSEUDO_SUBFOLDER, filename)
+                ))
+
+        codeinfo = datastructures.CodeInfo()
+        codeinfo.stdin_name = self.inputs.metadata.options.input_filename
+        codeinfo.stdout_name = self.inputs.metadata.options.output_filename
+        codeinfo.code_uuid = self.inputs.code.uuid
+
+        calcinfo = datastructures.CalcInfo()
+        calcinfo.codes_info = [codeinfo]
+        calcinfo.local_copy_list = local_copy_list
+        calcinfo.remote_copy_list = remote_copy_list
+
+        # Retrieve by default the output file and plot file
+        calcinfo.retrieve_list = []
+        calcinfo.retrieve_list.append(self.inputs.metadata.options.output_filename)
+        if self.inputs.metadata.options.keep_plot_file:
+            calcinfo.retrieve_list.append(self._FILEOUT)
+        else:
+            calcinfo.retrieve_temporary_list = [self._FILEOUT]
```

Comment:
@sphuber  - Maybe you have some insight - this doesn't seem to do what I would expect. 
Even without the if/else block, all files are retrieved, rather than just those specified, as if retrieve_list is being ignored.

Reply:
Well for a kickoff, I cannot even find now where you are telling `pp.x` to write to these files. They are blocked keywords, so the plugin should add them to the parameters, correct? Something like
```
parameters = self.inputs.parameters.get_dict()
parameters['INPUTPP']['filplot'] = self._FILPLOT
parameters['INPUTPP']['filout'] = self._FILEOUT
```
or am I missing something here?

Reply:
Yes, those variables are set right at the top of the class. and later the relevant keywords added to the  blocked list:
https://github.com/aiidateam/aiida-quantumespresso/blob/7a294c4be22ecaae62618d93aa77f9a7e6a362dd/aiida_quantumespresso/calculations/pp.py#L63-L66

I've added a `PpCalculation` test class. 

Retrieving the files works correctly now also.